docs/DeveloperGuide.adoc

```diff
@@ -137,130 +146,245 @@ image:BetterModelClassDiagram.png[]
+implement AccountMoney object, which consists of 2 objects:
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+****
+1. `Amount` indicated the value added/withdrew and it is displayed in the UI.
+2. `TimeStamp` indicated the date and time the amount of money was added/withdrawn.
+****
 
-[NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+Each `AccountMoney` object is unique because the `TimeStamp` will never be the same. Since it is not possible for the user to do a saving
+and a withdrawal at a single time.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+.Savings class diagram in Model component.
+image::SavingsAccountClassDiagram.png[width=500]
 
-image::UndoRedoState4.png[]
+==== *Adding to the Savings Account*
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+The user may want to add a new saving to his savings account. This can be done with the `save` command.
 
-image::UndoRedoState5.png[]
+The savings/withdrawals are stored in the SavingsAccount in the SavingsHistory object, which stores the savings/withdrawal
+as a list.
 
-The following activity diagram summarizes what happens when a user executes a new command:
+The current implementation for adding a new saving/withdrawal is done by overwriting the existing SavingsHistory, with a new
+SavingsHistory that includes past and new savings/withdrawals.
 
-image::CommitActivityDiagram.png[]
+The command is read as a text string from the command box in the UI and then is executed by calling MainWindow#executeCommand(),
+which passes this string (named commandText) to the Logic component by calling Logic#execute(Model model).
 
-==== Design Considerations
+The sequence diagram for interactions between the Logic, Model and Storage components when a user executes the `save` command is shown below.
 
-===== Aspect: How undo & redo executes
+.Sequence diagram for sample save command
+image::SaveSequenceDiagram.png[]
```

Comment:
Technical details:
- For this save sequence diagram, perhaps you can consider removing the participant SavingsAccount as it does not seem to be used. 
- For LogicManager, the activation bar seems to cut off prematurely - before Storage#saveSavingsAccount (it should deactivate at the bottom arrow instead).
- Storage should be StorageManager instead, as Storage is an interface.

Additionally, maybe you can consider splitting this sequence diagram (put a reference frame over LogicManager, Storage, and JsonStorageManager and include the storage components in the separate diagram).